packages/dev/core/src/Shaders/screenSpaceReflection.fragment.fx

```diff
@@ -174,3 +196,355 @@ void main()
+                return;
+            #endif
+        } else {
+            vec2 texSize = gl_FragCoord.xy / vUV;
+            info = getReflectionInfo2DRayMarching(reflected + jitt, position, texSize);
+        }
+    #else // if left handed scene
+        if (position.z > distanceFade || distanceFade == 0.0){ // no need to compute reflection, the point we are evaluating is further than the distanceFade
+            #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+                info.coords = vUV;
+                info.visibility = 0.0;
+                info.miss = true;
+                info.visibilityBackup = 1.0;
+            #else
+                gl_FragColor = texture2D(textureSampler, vUV); 
+                return;
+            #endif
+        } else {
+            vec2 texSize = gl_FragCoord.xy / vUV;
+            info = getReflectionInfo2DRayMarching(reflected + jitt, position, texSize);
+
+        }
+    #endif
+
+    // *************** Get reflection color ***************
+
+    vec3 reflectedColor;
+    
+    #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+        if (backupOnlyWhenTooSpecular && dot(spec, vec3(1.0))/3.0 > 0.9) {
+            info.visibility = 0.0;
+            info.visibilityBackup = 1.0;
+        }
+        // compute reflection in view space and then come back to world space
+        vec3 coord = vec3( inverse(view) * vec4(reflected, 0.0));
+
+        #ifdef BACKUP_TEXTURE_PROBE
+            coord.y *= -1.0;
+        #endif
+            
+        #ifdef RIGHT_HANDED_SCENE
+            coord.z *= -1.0;
+        #endif
+        reflectedColor = textureCube(backUpSampler, coord + jitt).xyz * info.visibilityBackup;
+
+        if (!info.miss){
+            reflectedColor += texture2D(textureSampler, info.coords).xyz * info.visibility;
+        }
+    #else 
+        if (backupOnlyWhenTooSpecular && dot(spec, vec3(1.0))/3.0 > 0.9) {
+            info.visibility = 0.0;
+            info.visibilityBackup = 0.0;
+        }
+        if (info.miss){
+            gl_FragColor = texture2D(textureSampler, vUV);
+            return;
+        } else {
+            reflectedColor = texture2D(textureSampler, info.coords).xyz;
+        }
+    #endif 
+    
+    //  *********************** Shading *******************************
+
+    // Fresnel
+    // "The specular map contains F0 for dielectrics and the reflectance value for raw metal"
+    vec3 F0 = spec;
+  
+    vec3 reflectionCoeff = fresnelSchlick(max(dot(unitNormal, -unitPosition), 0.0), F0) // https://lettier.github.io/3d-game-shaders-for-beginners/fresnel-factor.html
+                            * (info.visibility + info.visibilityBackup); 
+    reflectionCoeff = clamp(vec3(pow(reflectionCoeff.x * strength, falloffExponent - 2.0), pow(reflectionCoeff.y * strength, falloffExponent - 2.0), pow(reflectionCoeff.z * strength, falloffExponent - 2.0)), 0.0, 1.0); // -2 to compensate for default value (3.0)
+

```

Comment:
Why not `pow(reflectionCoeff * strength, fallOffExponent - 2.0)`?

Also why compensating for the `3` value in the shader? It's not a problem to have a default value of 3 as long as it is documented, and the user can change to smaller or any value they want during the setup. So, `pow(reflectionCoeff * strength, fallOffExponent)` would be better and more logical to me.

Reply:
To me, choosing 3 as default value doesn't make sense. However, it was the default value of the first version of SSR and I choose not to change this default value to avoid regression..
I guess it was used to compensate for the mistake in FresnelSchlick call (in the first version this value is always 1.0)

Reply:
I'm not sure to understand, in the existing version the line is `float reflectionMultiplier = clamp(pow(spec * strength, reflectionSpecularFalloffExponent) * screenEdgefactor * reflected.z, 0.0, 0.9);` so we are using `reflectionSpecularFalloffExponent` directly?

Regarding the backward compatibility, I wonder if it would not be easier to simply separate the implementations and have a `ScreenSpaceReflection2` class, the same way we have SSAO / SSAO2... That way we don't change anything of the existing typescript / shader code and we create new files instead.

What do you think @deltakosh / @sebavan?

Reply:
To me, reflectionSpecularFalloffExponent and strength are used for artistic purpose only. Thus, default value shouldn't affect the reflection coefficient based on Fresnel formula.

The Fresnel formula already takes into account the specularity (through F0). 
The existing version miss a minus before the unitPosition when calling fresnelSchlick function. So, this function always returns 1. This is why, I guess, the reflection multiplier is based on spec and uses 3.0 as reflectionSpecularFalloffExponent default value. 


Regarding backwardCompatibility, I defined the backwardCompatibility flag as mentioned in this thread : https://forum.babylonjs.com/t/ssr-when-i-rotated-a-mesh-the-reflection-shows-wrong-and-connected-to-the-mesh/30915/19. But I can create a new class with new files instead.

Reply:
Yes, the missing minus sign is a bug and the default of 3 value is not very sound.

That's why I think a separate implementation would be better. That would also be better for tree shaking.

Let's see what @deltakosh think about this (I think @sebavan is in holidays).

Reply:
Okay !
I'll create a new class and separate both versions.